torchgeo/datasets/splits.py

```diff
@@ -295,11 +296,12 @@ def time_series_split(
 
     .. versionadded:: 0.5
     """
-    x, y, t = dataset.bounds
+    _, _, t = dataset.bounds
 
     totalt = t.stop - t.start
 
     if all(isinstance(x, int | float) for x in lengths):
+        lengths = cast(Sequence[float], lengths)
```

Comment:
Mypy cannot understand the guarantees provided by the previous line, need to give it a helping hand